src/nat/mapped_tcp_socket/mod.rs

```diff
@@ -104,30 +99,13 @@ where
             token,
             socket: Some(socket),
             igd_children,
-            stun_children: HashSet::with_capacity(mc.peer_stuns().len()),
             mapped_addrs,
             timeout: core.set_timeout(Duration::from_secs(TIMEOUT_SEC), CoreTimer::new(token, 0)),
             finish: Some(finish),
             phantom: PhantomData,
         }));
 
-        // Ask Stuns
-        for stun in mc.peer_stuns() {
```

Comment:
If you don't do this how will you get your public address to tell others where to connect ? (Orthogonal to NAT traversal, e.g., even in case of direct-connections how are you informing the peers about where they should connect to you ? The tests probably pass as they fallback to loopback due to `0.0.0.0` nature of the addresses just now ?)

Reply:
Right.. ok, I'll bring this back.

Reply:
... for now, until p2p is not integrated yet :)